include/decoration.h

```diff
@@ -125,6 +125,47 @@ enum DecoId {
     /*120*/ DECOR_REGISTEEL_DOLL
 };
 
+enum DecorPerm {
+    /*
+     * The nomenclature here describes collision and placement permissions, in that order.
+     */
+    DECORPERM_SOLID_FLOOR,
+    DECORPERM_PASS_FLOOR,
+    DECORPERM_BEHIND_FLOOR,
+    DECORPERM_NA_WALL,
+    DECORPERM_SOLID_MAT
+};
+
+enum DecorShape {
+    /*
+     * Width-x-height
+     */
+    DECORSHAPE_1x1,
+    DECORSHAPE_2x1,
+    DECORSHAPE_3x1, // unused
+    DECORSHAPE_4x2,
+    DECORSHAPE_2x2,
+    DECORSHAPE_1x2,
+    DECORSHAPE_1x3, // unused
+    DECORSHAPE_2x4,
+    DECORSHAPE_3x3,
+    DECORSHAPE_3x2
+};
+
+enum DecoCat {
```

Comment:
Should be named DecorCat (or better, DecorationCategory) to keep consistent with the values of the enum.